packages/venia-ui/lib/components/CheckoutPage/checkoutPage.js

```diff
@@ -0,0 +1,238 @@
+            >
+                {'Login and Checkout Faster'}
+            </Button>
+        </div>
+    );
+};
+
+const GreetingMessage = props => {
+    const { isGuestCheckout, classes } = props;
+
+    return (
+        <div className={classes.heading_container}>
+            <h1 className={classes.heading}>
+                {isGuestCheckout ? 'Guest Checkout' : 'Review and Place Order'}
+            </h1>
+        </div>
+    );
+};
+
+const EmptyCartMessage = props => {
+    const { classes, isGuestCheckout } = props;
+
+    return (
+        <div className={classes.empty_cart_container}>
+            <div className={classes.heading_container}>
+                <h1 className={classes.heading}>
+                    {isGuestCheckout ? 'Guest Checkout' : 'Checkout'}
+                </h1>
+            </div>
+            <h3>There are no items in your cart.</h3>
+        </div>
+    );
+};
+
+export default props => {
+    const { classes: propClasses } = props;
+    const talonProps = useCheckoutPage({
+        createCartMutation: CREATE_CART_MUTATION,
+        getCartDetailsQuery: GET_CART_DETAILS_QUERY
+    });
+    const [
+        {
+            isGuestCheckout,
+            isCartEmpty,
+            shippingInformationDone,
+            shippingMethodDone,
+            paymentInformationDone,
+            orderPlaced
+        },
+        {
+            handleSignIn,
+            setShippingInformationDone,
+            setShippingMethodDone,
+            setPaymentInformationDone,
+            placeOrder
+        }
+    ] = talonProps;
+
+    const classes = mergeClasses(defaultClasses, propClasses);
+
+    const guestCheckout = () => {
+        if (isGuestCheckout) {
+            return (
+                <GuestCheckoutOptions
+                    classes={classes}
+                    handleSignIn={handleSignIn}
+                />
+            );
+        } else {
+            return null;
+        }
+    };
+
+    const priceAdjustments = () => {
+        const showPriceAdjustments =
+            shippingInformationDone &&
+            shippingMethodDone &&
+            !paymentInformationDone;
+
+        if (showPriceAdjustments) {
+            return (
+                <div className={classes.price_adjustments_container}>
+                    <PriceAdjustments />
+                </div>
+            );
+        } else {
+            return null;
+        }
+    };
+
+    const itemsReview = () => {
+        const showItemsReview =
+            shippingInformationDone &&
+            shippingMethodDone &&
+            paymentInformationDone;
+
+        if (showItemsReview) {
+            return (
+                <div className={classes.items_review_container}>
+                    <ItemsReview />
+                </div>
+            );
+        } else {
+            return null;
+        }
+    };
+
+    const showOrderConfirmation = isCartEmpty && orderPlaced;
+
+    const orderConfirmation = useMemo(() => {
```

Comment:
Missed a memo :/

Reply:
Left it on purpose, wanted to talk to the team and jimmy about it. Removed everything else.

Reply:
Done. Changed.